src/EditorFeatures/CSharpTest/QuickInfo/DiagnosticAnalyzerQuickInfoSourceTests.cs

```diff
@@ -0,0 +1,220 @@
+                : null;
+            await TestAsync(
+@$"
+using System.Diagnostics.CodeAnalysis;
+using SM = System.Diagnostics.CodeAnalysis.SuppressMessageAttribute;
+namespace T
+{{
+    public static class DiagnosticIds
+    {{
+        public const string IDE0051 = ""IDE0051"";
+    }}
+
+    {suppressMessageAttribute}
+    public class C
+    {{
+        private int _i;
+    }}
+}}
+", description, ImmutableArray<TextSpan>.Empty);
+        }
+        protected static async Task AssertContentIsAsync(TestWorkspace workspace, Document document, int position, string expectedDescription,
+            ImmutableArray<TextSpan> relatedSpans)
+        {
+            var info = await GetQuickinfo(workspace, document, position);
+            var description = info?.Sections.FirstOrDefault(s => s.Kind == QuickInfoSectionKinds.Description);
+            Assert.NotNull(description);
+            Assert.Equal(expectedDescription, description.Text);
+            Assert.Collection(relatedSpans,
+                info.RelatedSpans.Select(actualSpan => new Action<TextSpan>(expectedSpan => Assert.Equal(expectedSpan, actualSpan))).ToArray());
+        }
+
+        private static async Task<QuickInfoItem> GetQuickinfo(TestWorkspace workspace, Document document, int position)
+        {
+            var diagnosticAnalyzerService = workspace.ExportProvider.GetExportedValue<IDiagnosticAnalyzerService>();
+            var sut = new CSharpDiagnosticAnalyzerQuickInfoProvider(diagnosticAnalyzerService);
+            var info = await sut.GetQuickInfoAsync(new QuickInfoContext(document, position, CancellationToken.None));
+            return info;
+        }
+
+        protected static async Task AssertNoContentAsync(TestWorkspace workspace, Document document, int position)
+        {
+            var info = await GetQuickinfo(workspace, document, position);
+            Assert.Null(info);
+        }
+
+        protected static async Task TestAsync(
+            string code,
+            string expectedDescription,
+            ImmutableArray<TextSpan> relatedSpans,
+            CSharpParseOptions parseOptions = null)
+        {
+            using var workspace = TestWorkspace.CreateCSharp(code, parseOptions);
+            var analyzerReference = new AnalyzerImageReference(ImmutableArray.Create<DiagnosticAnalyzer>(
+                new CSharpCompilerDiagnosticAnalyzer(),
+                new CSharpRemoveUnusedMembersDiagnosticAnalyzer()));
+            workspace.TryApplyChanges(workspace.CurrentSolution.WithAnalyzerReferences(new[] { analyzerReference }));
+
+            var diagnosticAnalyzerService = workspace.ExportProvider.GetExportedValue<IDiagnosticAnalyzerService>();
+            var analyzer = (diagnosticAnalyzerService as IIncrementalAnalyzerProvider).CreateIncrementalAnalyzer(workspace);
+            await analyzer.AnalyzeProjectAsync(workspace.CurrentSolution.Projects.Single(), semanticsChanged: true, InvocationReasons.Reanalyze, CancellationToken.None);
```

Comment:
Is this call required?

Reply:
It's no longer needed. I remove it in 99f97be5.